JPetTaskIO/JPetTaskIO.cpp

```diff
@@ -211,12 +226,22 @@ bool JPetTaskIO::createOutputObjects(const char* outputFilename)
 {
   fWriter = new JPetWriter( outputFilename );
   assert(fWriter);
-  if (fSubTask) {
-    auto task = dynamic_cast<JPetUserTask*>(fSubTask.get());
-    task->setStatistics(fStatistics.get());
+  if (!fSubTasks.empty()) {
+    int i = 0;
+    for (auto fSubTask = fSubTasks.begin(); fSubTask != fSubTasks.end(); fSubTask++) {
+      auto task = dynamic_cast<JPetUserTask*>(fSubTask->get());
+      fSubTasksStatistics[task->getName()
+                          + std::string(" subtask ")
+                          + std::to_string(i)
+                          + std::string(" stats")]
+        = std::move(std::unique_ptr<JPetStatistics>(new JPetStatistics(*fStatistics)));
+
```

Comment:
I have pulled your changes and tried to run one of the examples, but it seems that the JPetStatistics pointer passed to the subtask here is null. If I add assert(fSubTasksStatistics[task->getName()].get()) before task->SetStatistics(...), the assertion fails.

Shouldn't the key which you use here be the same as above, i.e. not only task->getName() but with everything you append to it above?

Reply:
This comment mostly applies to the line right below the marked ones. Sorry for the confusion.

Reply:
Yes, I fixed that, I added to subtask name information about number of subtask to distinguish stats from different subtasks, but with same name, and forgot to change that in setStatistics